dedupe/ckan_api.py

```diff
@@ -156,12 +156,13 @@ def get_datasets_in_collection(self, package_id):
             return search_result['results']
         return None
 
-    def get_datasets(self, organization_name, identifier, start=0, rows=1000,
+    def get_datasets(self, organization_name, guid, start=0, rows=1000,
                      is_collection=False):
         filter_query = \
-            'identifier:"%s" AND organization:"%s" AND type:dataset' % \
-            (identifier, organization_name)
+            'guid:"%s" AND organization:"%s" AND type:dataset' % \
+            (guid, organization_name)
```

Comment:
Leave the identifier here as identifier. Add a new optional parameter similar to sort_order called identifier_type that defaults to identifier, and then use that parameter in place of the first identifier on line 162.